orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -843,3 +853,368 @@ def _update_onnx_model_initializers(self, state_tensors):
         for w_i in replace_indices:
             del self._onnx_model.graph.initializer[w_i]
         self._onnx_model.graph.initializer.extend(new_weights)
+
+    def _extract_model_states(self, state_dict, pytorch_format):
+        """Extract model states from the training session and load into the state_dict"""
+
+        model_states = self._training_session.get_model_state(include_mixed_precision_weights=False)
+        state_dict[_utils.state_dict_model_key()] = {}
+
+        # extract trained model weights from the training session
+        for precision in model_states:
+            state_dict[_utils.state_dict_model_key()][precision] = {}
+            for model_state_key in model_states[precision]:
+                if pytorch_format:
+                    state_dict[_utils.state_dict_model_key()][precision][model_state_key] = \
+                        torch.from_numpy(model_states[precision][model_state_key])
+                else:
+                    state_dict[_utils.state_dict_model_key()][precision][model_state_key] = \
+                        model_states[precision][model_state_key]
+
+        # extract untrained (frozen) model weights
+        precision_map = _utils.state_dict_precision_map()
+        for node in self._onnx_model.graph.initializer:
```

Comment:
can we avoid checking the type of frozen weights and return them nonetheless? We have had a model have a non-trainable buffer of type bool/uint8 or any other numeric type.

Reply:
yes, I think this would make sense. I will put all of them under the bucket ```fp32``` so as to indicate that these weights are not trained and to the full precision model.